src/components/FormBuilder.jsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useReducer, useState } from "react";
+import reducer, { getInitialState, updateFieldValue, updateFieldError, setErrors, reset } from "../reducers/form";
+import { CheckboxGroup, Input, InputRadioGroup, Select, TextArea } from "./FormFields";
+import definedValidations from "../helpers/validators";
+import { INPUT_TYPES } from "../constants/form";
+
+const defaultArr = [];
+
+const definedValidationsArr = Object.keys(definedValidations);
+
+const renderField = field => field;
+
+function FormBuilder({ config }) {
+  const {
+    fields,
+    onSubmit,
+    wrapperClass,
+    id: formId,
+    title: { text: heading, class: headingClass },
+    submitBtn: { id: submitBtnId, class: submitBtnClass, content: submitBtnContent }
+  } = config;
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [state, dispatch] = useReducer(reducer, fields, getInitialState);
+
+  const { values, errors } = state;
+
+  useEffect(() => {
+    if (isSubmitting) {
+      const hasErrors = Object.values(errors).some(Boolean);
+
+      if (!hasErrors) {
+        onSubmit(values);
+        dispatch(reset(fields));
```

Comment:
Why are we sending fields in reset ?

Reply:
after submit, the form should be filled with initial values